### PR TITLE
[chore] update sapm-proto to 0.16.0

### LINKED ISSUE #35626: [receiver/sapm] Remove dependency on opencensus
### Component(s)

receiver/sapm

### Describe the issue you're reporting

➜  sapmexporter git:(rm-opencensus-goleak) go mod why go.opencensus.io
# go.opencensus.io
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter
github.com/signalfx/sapm-proto/client
go.opencensus.io/stats
go.opencensus.io/stats.test
go.opencensus.io/trace
go.opencensus.io/internal
go.opencensus.io

This dependency is no longer warranted. It should be removed so we no longer import opencensus into contrib.

### PR DESCRIPTION
**Pull request by atoulme**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35626

### CODE REVIEW

File: exporter/sapmexporter/go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/sapmexporter
 
-go 1.22.0
+go 1.23
+
+toolchain go1.23.2
```
Comment (atoulme) [inline exporter/sapmexporter/go.mod:5]:
```suggestion
go 1.22.0
```

File: testbed/go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/testbed
 
-go 1.22.0
+go 1.23
+
+toolchain go1.23.2
```
Comment (atoulme) [inline testbed/go.mod:5]:
```suggestion
go 1.22
```

File: receiver/sapmreceiver/go.mod [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sapmreceiver
 
-go 1.22.0
+go 1.23
+
+toolchain go1.23.2
```
Comment (atoulme) [inline receiver/sapmreceiver/go.mod:5]:
```suggestion
go 1.22.0
```

File: exporter/sapmexporter/metadata.yaml [hunk 4]
```diff
@@ -11,8 +11,4 @@ status:
 
 tests:
   expect_consumer_error: true
-  goleak:
-    ignore:
-      top:
-        # See https://github.com/census-instrumentation/opencensus-go/issues/1191 for more information.
-        - "go.opencensus.io/stats/view.(*worker).start"
\ No newline at end of file
+  goleak:
```
Comment (bogdandrutu) [inline exporter/sapmexporter/metadata.yaml:14]: No need to have it as empty, enabled by default. Also add the new line.

**Review by bogdandrutu — APPROVED** [review]

File: exporter/sapmexporter/metadata.yaml (hunk 4 above)
Comment (atoulme) [reply exporter/sapmexporter/metadata.yaml]: thanks, done.

**Review by pjanotti — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**